### DISCUSSION TITLE
errors while import project using gradle

### DISCUSSION THREAD

**Discussion by tgkei** [scalameta/metals#2994] [Q&A] [answered]
I try to use `coc-metals` for my scala project, but it keeps throws `Import project failed, no functionality will work. See the logs for more details` when I run `Build tool selection` Below is the error log

<img width="2560" alt="스크린샷 2021-07-27 오후 5 25 06" src="https://user-images.githubusercontent.com/37530453/127121851-d00744c3-531e-4c33-ac80-7e3510735462.png">

As you guys can see above, it throws error while running `gradle bloopInstall`. Does anyone know how to solve this problem?
My `gradle` version is 7.1.1

**Comment by ckipp01** [comment]
Hey @tgkei is the project public by chance? I'm not extremely familiar with Gradle, but from looking at the error it seems to trip up on the `mainClass` setting.

**Reply by tgkei** [reply]
@ckipp01 here is project github link : https://github.com/tgkei/sparktest

it's nothing but only to learn `coc-metals`

**Reply by ckipp01** [reply]
I transfered this over to the Metals repo since I'm not `coc-metals` specific. Again I'm pretty unfamiliar with Gradle, but I'm unable to actually get the `bloopInstall` to work even manually using Bloop, but I could be doing something wrong. Putting this in here in case one of the other know more about the Bloop + Gradle integration. If not, I'll have to dig a bit deeper into this to see what's going on. In the mean time, if you're not tied to Gradle, you should be able to use sbt or Mill for a very minimal project like this, which will provide you with a smoother experience.

**Reply by tgkei** [reply]
alright. appreciate for your help

**Comment by tgkei** [comment, marked as answer]
It was nothing related to `coc-metals`. It's all about the way to set `mainClass` using `gradle`. When using gradle version 7.1.1, `gradle.init` set `mainClass` using `mainClass = 'sparktest.App'`, but it should be changed to `setMainClass("sparktest.App")` and it worked for me

**Reply by ckipp01** [reply]
Ah, great. Glad you got it figured out!
